File: frontend/src/api/reviewComment.ts
import { client } from './client'

// 审核批注：审核员把问题钉在具体的 帧 + track 上，标注员点一下就跳到问题点，
// 逐条修复并标记「已修复」。仍有未处理批注时后端拒绝重新提交（B3.1）。
export interface ReviewComment {
  id: string
  task_id: number
  frame?: number
  track_id?: number
  time_ms?: number
  body: string
  status: 'open' | 'resolved'
  author_id: number
  author_name?: string
  resolved_by?: number
  resolved_at?: string
  created_at: string
}

export interface CommentAnchor {
  frame?: number
  track_id?: number
  time_ms?: number
}

export const reviewCommentApi = {
  async list(taskId: number): Promise<ReviewComment[]> {
    const { data } = await client.get<{ items: ReviewComment[] }>(`/tasks/${taskId}/review-comments`)
    return data.items ?? []
  },

  async create(taskId: number, anchor: CommentAnchor, body: string): Promise<ReviewComment> {
    const { data } = await client.post<ReviewComment>(`/tasks/${taskId}/review-comments`, { ...anchor, body })
    return data
  },

  async setResolved(taskId: number, commentId: string, resolved: boolean): Promise<void> {
    await client.patch(`/tasks/${taskId}/review-comments/${commentId}`, { resolved })
  },

  async remove(taskId: number, commentId: string): Promise<void> {
    await client.delete(`/tasks/${taskId}/review-comments/${commentId}`)
  },
}

File: backend/internal/model/payload/review_comment.go
package payload

import "time"

// CollReviewComment holds reviewer comments anchored to a point in the asset.
const CollReviewComment = "mm_review_comments"

// Review comment status values.
const (
	ReviewCommentOpen     = "open"
	ReviewCommentResolved = "resolved"
)

// ReviewComment is a reviewer's note pinned to a specific place in the asset,
// so a rejected task tells the annotator *where* the problem is rather than
// just that there is one (执行方案-02 B3.1).
//
// The anchor fields are all optional and interpreted per modality:
//   - video: Frame (+ TrackID when the problem is one object)
//   - audio: TimeMs
//   - image: TrackID doubles as the shape index
//
// A comment with no anchor is a whole-task remark. Resolution is the
// annotator's action — the reviewer files it, the annotator closes it — so a
// task can be re-submitted with every comment accounted for.
type ReviewComment struct {
	ID        string `json:"id"`
	TaskID    uint   `json:"task_id"`
	DatasetID uint   `json:"dataset_id"`
	AssetID   uint   `json:"asset_id"`

	Frame   *int     `json:"frame,omitempty"`
	TrackID *int     `json:"track_id,omitempty"`
	TimeMs  *float64 `json:"time_ms,omitempty"`

	Body string `json:"body"`

	Status     string     `json:"status"` // open | resolved
	ResolvedBy *uint      `json:"resolved_by,omitempty"`
	ResolvedAt *time.Time `json:"resolved_at,omitempty"`

	AuthorID   uint   `json:"author_id"`
	AuthorName string `json:"author_name,omitempty"`

	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`
}
